test/example/c-api-kitchen-sink.c

```diff
@@ -870,7 +870,7 @@ void test_core() {
   // Memory. One per module
 
   const char* segments[] = { "hello, world", "I am passive" };
-  int8_t segmentPassive[] = { 0, 1 };
+  bool segmentPassive[] = { false, true };
```

Comment:
are these changes necessary - do compilers not automatically convert these silently?

if so then it's a breaking change and so I think it's worth mentioning in the changelog.

Reply:
They silently convert variables but not arrays, which is why this was failing: https://github.com/WebAssembly/binaryen/runs/2076841915?check_suite_focus=true#step:17:18642

Would a sentence like "Boolean values in the C API now should use `bool` instead of `int`." be enough, or do you want to list all the affected functions?

Reply:
I think a short sentence is good enough, no need for a full list.

But this does make me worry. If arrays are common usage, then this could be annoying for users. I suspect it would be fine as most users actually use some kind of bindings to another language anyhow (not many write compilers in C). Curious to hear your thoughts and others'.

Reply:
I don't have see if errors or warnings when I do `bool foo[] = {0, 0, 1, 1};`... at last not from `clang -Wall`

Reply:
Actually if this is compiled in C, there's no problem. The problem occurs when you compile it in C++. You get a narrowing conversion warning like:
> warning: narrowing conversion of ‘x’ from ‘int8_t’ {aka ‘signed char’} to ‘bool’ [-Wnarrowing]

In any case, I don't totally understand why this could be annoying. You mean because people are used to writing 0 and 1 instead of false and true? Or because they have to type more characters? In any case, I don't personally find it annoying. Also as you say, most will use bindings but even if they didn't it is fine for me. Although maybe there are others who are more picky with this sort of thing. Lets wait for more opinions then.

Reply:
I don't see that warning with clang of gcc.. maybe a windows thing?

```
$ cat test.cpp 
#include <stdbool.h>

bool foo[] = {0, 0, 1, 1};
$ g++ -Wall test.cpp -c
$ clang++ -Wall test.cpp -c
```

Reply:
Regardless, seems like a pretty minimal breakage that would be easy for folks to fix (Also, its only a warning not an error)

Reply:
@sbc100 So, assigning `bool foo[] = {0, 0, 1, 1};` is fine. The conversion is done per value, which is ok since you can do `bool x = 1;`. The problem is for example with this:
```
bool foo(uint8_t x) {
    bool arr[] = { x, x};
    return arr[0];
}
```

Reply:
I sort of wandered away from the initial problem. I am sorry for this. The initial problem is not even C++ related, because the API tests are compiled with gcc. The initial problem is this: I changed the header file to accept `bool *` for passive segments field. But the test is calling it with an `int8_t *`. So the following example comes close to understanding the compilation issue:
```
#include <stdbool.h>
#include <stdint.h>

void FunctionExpectsBoolArray(bool*);

void foo(void) {
    int8_t arr[] = { 0, 1 };
    FunctionExpectsBoolArray(arr);
}
```

If you try with `gcc`:
```
gcc -Wall -c ~/tmp/bool4.c 
/home/pmatos/tmp/bool4.c: In function ‘foo’:
/home/pmatos/tmp/bool4.c:8:30: warning: passing argument 1 of ‘FunctionExpectsBoolArray’ from incompatible pointer type [-Wincompatible-pointer-types]
    8 |     FunctionExpectsBoolArray(arr);
      |                              ^~~
      |                              |
      |                              int8_t * {aka signed char *}
In file included from /home/pmatos/tmp/bool4.c:1:
/home/pmatos/tmp/bool4.c:4:31: note: expected ‘_Bool *’ but argument is of type ‘int8_t *’ {aka ‘signed char *’}
    4 | void FunctionExpectsBoolArray(bool*);
      |                               ^
```

Replying to @sbc100 , if you change the declaration to `bool arr[] = { 0, 1 };` it works, as you rightly pointed out.